src/main/java/com/sobytylnik/hello/InMemoryProfileRepository.java

```diff
@@ -0,0 +1,69 @@
+package com.sobytylnik.hello;
+
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.sobytylnik.hello.ExceptionListPackage.EntityAlreadyExistsException;
+import com.sobytylnik.hello.ExceptionListPackage.EntityNotFoundException;
+
+
+public class InMemoryProfileRepository implements ProfileRepository {
+
+
+    ConcurrentHashMap<Long, Profile> ProfileMap = new ConcurrentHashMap<>();
+
+
+@Override
+    public Optional<Profile> findById(Long id) {
+
+        if (ProfileMap.containsKey(id)) {
+            Profile a = ProfileMap.get(id);
+
+
+            return Optional.of(a);
+        }
+        return Optional.empty();
+    }
+
+
+
+    /**
+     * @throws EntityAlreadyExistsException
+     */
+    @Override
+    public Profile save(Profile profile) {

```

Comment:
Add logic for ID generation and injection into profile via setter.
Test this logic.